### PR TITLE
JBDS-4080  Avoid to redownload everything on each install

### PR DESCRIPTION
**Pull request by dgolovin**
Fix adds verification for downloaded files in temp folder and
if sha is the same installer uses those files instead of downloading
them again.

### CODE REVIEW

File: browser/model/installable-item.js [hunk 1]
```diff
@@ -113,6 +116,38 @@ class InstallableItem {
     success();
   }
 
+  checkAndDownload(dl,url,sha,user,pass) {
```
Comment (jrichter1) [inline browser/model/installable-item.js]: I believe we should provide unit test with new methods. Please add tests for this method. And if you could remove any unnecessary debug messages, that would be smashing.

File: browser/model/installable-item.js [hunk 2]
```diff
@@ -113,6 +116,38 @@ class InstallableItem {
...
+          this.startDownload(dl,url,sha,user,pass);
+        }
+      });
+    } else {
+      console.log("here1------------------------------------------------------------------------");
+      this.startDownload(dl,url,sha,user,pass);
+    }
+  }
+
+  startDownload(dl,url,sha,user,pass) {
```
Comment (jrichter1) [inline browser/model/installable-item.js]: Please provide unit tests as well for this method.

**Comment by dgolovin** [comment]
Refactored download a little bit to get rid of code duplication and added tests for new methods.

**Comment by jrichter1** [comment]
Chick Norris Approves

**Pull request merged**